packages/native/examples/with-controlled-props/App.js

```diff
@@ -0,0 +1,222 @@
+import React, { useState, useRef } from 'react';
+import {
+  SearchBase,
+  SearchComponent,
+  SearchBox
+} from '@appbaseio/react-native-searchbox';
+import { AntDesign } from '@expo/vector-icons';
+import {
+  StyleSheet,
+  Text,
+  View,
+  Platform,
+  StatusBar,
+  ActivityIndicator,
+  FlatList,
+  Image,
+  SafeAreaView
+} from 'react-native';
+
+const renderResultItem = ({ item }) => {
+  return (
+    <View style={styles.itemStyle}>
+      <Image
+        style={styles.image}
+        source={{
+          uri: item.image
+        }}
+        resizeMode="contain"
+      />
+      <View style={{ flex: 1 }}>
+        <Text style={styles.textStyle}>{item.original_title}</Text>
+        <Text style={styles.textStyle}>by {item.authors}</Text>
+        <View style={styles.star}>
+          {Array(item.average_rating_rounded)
+            .fill('x')
+            .map((i, index) => (
+              <AntDesign
+                key={item._id + `_${index}`}
+                name="star"
+                size={24}
+                color="gold"
+              />
+            ))}
+          <Text style={styles.rating}>({item.average_rating} avg)</Text>
+        </View>
+        <Text>Pub {item.original_publication_year}</Text>
+      </View>
+    </View>
+  );
+};
+
+const renderItemSeparator = () => {
+  return (
+    // Flat List Item Separator
+    <View style={styles.itemSeparator} />
+  );
+};
+
+export default function App() {
+  const [dataSource, setDataSource] = useState([]);
+  const [resetPagination, setResetPagination] = useState(false);
+  const [text, setText] = useState('');
+  const stateRef = useRef();
+  stateRef.current = dataSource;
+  const stateRefQuery = useRef();
+  stateRefQuery.current = resetPagination;
+  const setResults = results => {
+    if (stateRefQuery.current) {
+      // Reset paginated data source
+      setDataSource(results.data);
+      setResetPagination(false);
+    } else {
+      setDataSource([...stateRef.current, ...results.data]);
+    }
+  };
+  return (
+    <SafeAreaView style={styles.container}>
+      <SearchBase
+        index="good-books-ds"
+        credentials="a03a1cb71321:75b6603d-9456-4a5a-af6b-a487b309eb61"
+        url="https://appbase-demo-ansible-abxiydt-arc.searchbase.io"
+        appbaseConfig={{
+          recordAnalytics: true,
+          enableQueryRules: true,
+          userId: 'jon@appbase.io',
+          customEvents: {
+            platform: 'ios',
+            device: 'iphoneX'
+          }
+        }}
+      >
+        <SearchBox
+          id="search-component"
+          dataField={[
+            {
+              field: 'original_title',
+              weight: 1
+            },
+            {
+              field: 'original_title.search',
+              weight: 3
+            }
+          ]}
+          onValueSelected={value => {
+            setResetPagination(true);
+          }}
+          value={text}
+          onChange={(value, searchComponent) => {
```

Comment:
@anik-ghosh-au7 If you're using hooks, then this example is wrong. It would trigger the query for old query value. I've mentioned in notion card that how to use with hooks.

Reply:
@bietkul updated the changes